genesyscloud/resource_genesyscloud_script.go

```diff
@@ -31,6 +31,26 @@ func getAllScripts(ctx context.Context, clientConfig *platformclientv2.Configura
 			break
 		}
 		for _, script := range *scripts.Entities {
+			//We only want scripts that are published.
+			_, resp, err := scriptsAPI.GetScriptsPublishedScriptId(*script.Id, "")
```

Comment:
Added code here to check whether a script is published.  We only want to return the published scripts and not unpublished scripts